sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -264,6 +264,11 @@ protected string GetOptionalVariable(string name)
                 value = Environment.GetEnvironmentVariable(name);
             }
 
+            if (value == null)
```

Comment:
Why do we need to support the "AZURE_" prefix?

Reply:
so that it can read the regular defaultcredential env variables locally.  They are the exact same variables just with AZURE_ prepended so I added that to the check list